src/apis/googleapi.cpp

```diff
@@ -49,6 +49,12 @@ QString GoogleAPI::translate(const QString &input,
 
     // If error.
     if (replyJsonObject.contains("error")) {
+        QJsonObject errorJsonObject = replyJsonObject["error"].toObject();
+        int errorCode = errorJsonObject["code"].toInt();
+        std::string errorMessage = errorJsonObject["message"]
+                .toString()
+                .toStdString();
+        DLOG(INFO) << "Error " + std::to_string(errorCode) + ": " + errorMessage;
```

Comment:
I'd suggest to use a single exception handler instead of littering code with logs. Also once you start reading the stack trace of `InvalidArgumentException`, you wont notice this log line. What you'll see will be `InvalidArgumentException` with no context. Due to this you'll have to look for additional log messages which will require a lot more time in order to get more details on what caused this issue.

Thats why `InvalidArgumentException` should have a message inside it.